data_model/src/evaluate.rs

```diff
@@ -1,105 +1,107 @@
 //! Implementations for Expression evaluation for different expressions.
-#![allow(
-    clippy::module_name_repetitions,
-    clippy::std_instead_of_core,
-    clippy::std_instead_of_alloc,
-    clippy::arithmetic_side_effects
-)]
-use eyre::Result;
-use iroha_data_model::{
+
+#[cfg(not(feature = "std"))]
+use alloc::{collections::BTreeMap, string::ToString};
+#[cfg(feature = "std")]
+use std::collections::BTreeMap;
+
+use crate::{
     expression::{prelude::*, Expression},
     isi::error::{InstructionExecutionFailure as Error, MathError},
     prelude::*,
-    query::error::FindError,
+    query::error::{FindError, QueryExecutionFailure},
 };
 
-use super::Evaluate;
-use crate::{prelude::ValidQuery, wsv::WorldStateView};
+/// Context of expression evaluation, holding (name, value) pairs for resolving identifiers
```

Comment:
Could you maybe expand on this? It's not obvious neither what are the identifiers we're talking about nor why we need them

Reply:
you are correct, it can be a pain to figure out what `Context` does. I've updated the docs now with a neat example